src/gazebo_ros_accel_sensor.cpp

```diff
@@ -0,0 +1,237 @@
+    imu_msg.orientation.z = 0;  //orientation.Z() + GuassianKernel(0,gaussian_noise);
+    imu_msg.orientation.w = 0;  //orientation.W() + GuassianKernel(0,gaussian_noise);
+
+    imu_msg.linear_acceleration.x = accelerometer_data.X() + GuassianKernel(0,gaussian_noise);
+    imu_msg.linear_acceleration.y = accelerometer_data.Y() + GuassianKernel(0,gaussian_noise);
+    imu_msg.linear_acceleration.z = accelerometer_data.Z() + GuassianKernel(0,gaussian_noise);
+
+    imu_msg.angular_velocity.x = 0; //gyroscope_data.X() + GuassianKernel(0,gaussian_noise);
+    imu_msg.angular_velocity.y = 0; //gyroscope_data.Y() + GuassianKernel(0,gaussian_noise);
+    imu_msg.angular_velocity.z = 0; //gyroscope_data.Z() + GuassianKernel(0,gaussian_noise);
+
+    //covariance is related to the Gaussian noise
+    double gn2 = gaussian_noise*gaussian_noise;
+    imu_msg.orientation_covariance[0] = -1.0; //gn2;
+    imu_msg.orientation_covariance[4] = 0;    //gn2;
+    imu_msg.orientation_covariance[8] = 0;    //gn2;
+    imu_msg.angular_velocity_covariance[0] =    gn2;
+    imu_msg.angular_velocity_covariance[4] =    gn2;
+    imu_msg.angular_velocity_covariance[8] =    gn2;
+    imu_msg.linear_acceleration_covariance[0] = gn2;
+    imu_msg.linear_acceleration_covariance[4] = gn2;
+    imu_msg.linear_acceleration_covariance[8] = gn2;
+
+    //preparing message header
+    imu_msg.header.frame_id = body_name;
+    imu_msg.header.stamp.sec = current_time.sec;
+    imu_msg.header.stamp.nsec = current_time.nsec;
+
+    //publishing data
+    imu_data_publisher.publish(imu_msg);
+
+    ros::spinOnce();
+  }
+
+  last_time = current_time;
+}
+
+double gazebo::GazeboRosAccelSensor::GuassianKernel(double mu, double sigma)
+{
+  // generation of two normalized uniform random variables
+  double U1 = static_cast<double>(rand_r(&seed)) / static_cast<double>(RAND_MAX);
+  double U2 = static_cast<double>(rand_r(&seed)) / static_cast<double>(RAND_MAX);
+
+  // using Box-Muller transform to obtain a varaible with a standard normal distribution
+  double Z0 = sqrt(-2.0 * ::log(U1)) * cos(2.0*M_PI * U2);
+
+  // scaling
+  Z0 = sigma * Z0 + mu;
+  return Z0;
+}
+
+bool gazebo::GazeboRosAccelSensor::LoadParameters()
+{
+  //loading parameters from the sdf file
+
+  //NAMESPACE
+  if (sdf->HasElement("robotNamespace"))
+  {
+    robot_namespace =  sdf->Get<std::string>("robotNamespace") +"/";
+    ROS_INFO_STREAM("<robotNamespace> set to: "<<robot_namespace);
+  }
+  else
+  {
+    std::string scoped_name = sensor->ParentName();
+    std::size_t it = scoped_name.find("::");
+
+    robot_namespace = "/" +scoped_name.substr(0,it)+"/";
+    ROS_WARN_STREAM("missing <robotNamespace>, set to default: " << robot_namespace);
+  }
+
+  //TOPIC
+  if (sdf->HasElement("topicName"))
+  {
+    topic_name =  robot_namespace + sdf->Get<std::string>("topicName");
+    ROS_INFO_STREAM("<topicName> set to: "<<topic_name);
+  }
+  else
+  {
+    topic_name = robot_namespace + "/imu_data";
```

Comment:
Maybe this should change on each of the plugins, otherwise both are publishing to the same topic.

Reply:
You are correct, that has to be changed. I will push a commit with a fix and  set same structure as the realsense camera (  ```robot_namespace + "/accel/sample"``` and  ```robot_namespace + "/gyro/sample"```)